javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -162,7 +167,7 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
      */
     public static <K extends Comparable<? super K>, V> TreeMap<K, V> ofAll(java.util.Map<? extends K, ? extends V> map) {
         Objects.requireNonNull(map, "map is null");
-        RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty();
+        RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty(new EntryComparator<>(Comparator.naturalOrder()));
```

Comment:
`K` **is** comparable because `K extends Comparable<? super K>`. We can't use natural comparator here. I believe this is correct fix:

```java
final Comparator<Tuple2<K, V>> comparator = new EntryComparator<>((Comparator<? super K> & Serializable) K::compareTo);
RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty(comparator);
```